protocols/v2/subprotocols/mining/src/lib.rs

```diff
@@ -203,7 +203,7 @@ impl Ord for Target {
     fn cmp(&self, other: &Self) -> core::cmp::Ordering {
         if self.tail == other.tail && self.head == other.head {
             core::cmp::Ordering::Equal
-        } else if self.head != other.head {
+        } else if self.tail != other.tail {
```

Comment:
Shouldn't this also check head equality?

```rs
} else if self.head == other.head && self.tail != other.tail {
```

(which is equivalent to `} else if self.head == other.head {`

Reply:
I don't think so. I think that the tail contain the most significant bits so if they are != then the head bits don't matter. We only need to compare the head bits if the tail bits are equal.

Reply:
Would be good to leave a comment above `pub struct Target` about whether this thing is big or little endian.

Reply:
Looks like U256 values in v2 are all little endian:
* https://github.com/stratum-mining/sv2-spec/blob/main/05-Mining-Protocol.md#532-openstandardminingchannel-client---server
* https://github.com/stratum-mining/sv2-spec/blob/main/03-Protocol-Overview.md#31-data-types-mapping

So that means `tail` bytes are the most significant. So your version is correct.